exporter/otlpexporter/config.go

```diff
@@ -1,61 +1,18 @@
-// Copyright The OpenTelemetry Authors
-// SPDX-License-Identifier: Apache-2.0
+// Code generated by github.com/atombender/go-jsonschema, DO NOT EDIT.
 
-package otlpexporter // import "go.opentelemetry.io/collector/exporter/otlpexporter"
+package otlpexporter
 
-import (
-	"errors"
-	"fmt"
-	"net"
-	"regexp"
-	"strconv"
-	"strings"
+import configretry "go.opentelemetry.io/collector/config/configretry"
+import exporterhelper "go.opentelemetry.io/collector/exporter/exporterhelper"
 
-	"go.opentelemetry.io/collector/component"
-	"go.opentelemetry.io/collector/config/configgrpc"
-	"go.opentelemetry.io/collector/config/configretry"
-	"go.opentelemetry.io/collector/exporter/exporterhelper"
-)
-
-// Config defines configuration for OTLP exporter.
+// Configuration parameters for the OTLP gRPC exporter.
 type Config struct {
-	exporterhelper.TimeoutSettings `mapstructure:",squash"`     // squash ensures fields are correctly decoded in embedded struct.
-	QueueConfig                    exporterhelper.QueueSettings `mapstructure:"sending_queue"`
-	RetryConfig                    configretry.BackOffConfig    `mapstructure:"retry_on_failure"`
-
-	configgrpc.ClientConfig `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
-}
-
-func (c *Config) Validate() error {
-	endpoint := c.sanitizedEndpoint()
-	if endpoint == "" {
-		return errors.New(`requires a non-empty "endpoint"`)
-	}
+	// RetryOnFailure corresponds to the JSON schema field "retry_on_failure".
+	RetryOnFailure *configretry.RetryOnFailure `mapstructure:"retry_on_failure,omitempty"`
 
-	// Validate that the port is in the address
-	_, port, err := net.SplitHostPort(endpoint)
-	if err != nil {
-		return err
-	}
-	if _, err := strconv.Atoi(port); err != nil {
-		return fmt.Errorf(`invalid port "%s"`, port)
-	}
+	// SendingQueue corresponds to the JSON schema field "sending_queue".
+	SendingQueue *exporterhelper.SendingQueue `mapstructure:"sending_queue,omitempty"`
 
-	return nil
+	// Timeout corresponds to the JSON schema field "timeout".
+	Timeout *exporterhelper.Timeout `mapstructure:"timeout,omitempty"`
 }
-
-func (c *Config) sanitizedEndpoint() string {
```

Comment:
how is this functionality achieved?

Reply:
At the moment it is not. The migration of `otlpexporter` and `otlphttpexporter` to Json schema isn't complete, due to a [blocker](https://github.com/open-telemetry/opentelemetry-collector/pull/10694#discussion_r1715704488) I hit with migrating squashed fields. I've been hoping we can find a solution to that blocker before we I fix the functionality of `otlpexporter` and `otlphttpexporter`.

But in general, we could put functions like this in `.go` helper files which are not generated by Json schema.